### PR TITLE
DGRSC JSON Logging

### PR DESCRIPTION
**Pull request by MichaelChung123**
# Description
Moving over from the npmlog/Morgan logging system and now we will be using JSON Format/Winston. 

## Types of changes
Added Winston JSON logging and changed all log calls to the new format.

## Checklist

- [X] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [X] I have checked that unit tests pass locally with my changes
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
PR Deployed at: https://dgrsc-dev.apps.silver.devops.gov.bc.ca/pr-77

File: app/package-lock.json [hunk 1]
```diff
@@ -14,13 +14,16 @@
         "axios-oauth-client": "^1.4.4",
         "axios-token-interceptor": "^0.2.0",
         "body-parser": "^1.19.0",
+        "carbone": "^3.2.3",
```
Comment (jujaga) [inline app/package-lock.json:17]: Why are we adding carbone to this app? The showcase app shouldn't need this at all.

File: app/app.js [hunk 2]
```diff
@@ -21,41 +19,12 @@ app.use(compression());
...
-      logFileStream.write(data);
-      done();
-    }
-  });
-  log.disableColor();
-  log.stream = teeStream;
-}
-
 // Print out configuration settings in verbose startup
 log.verbose('Config', utils.prettyStringify(config));
```
Comment (jujaga) [inline app/app.js:23]: These pre-existing log statements need to be reformatted - for this one specifically, we should not be pretty stringifying the config contents anymore as winston already handles object printing well. I'd consider either changing this to the following, or outright removing it (no function attribute as this is a top level action; set to **debug** as this is likely to print out sensitive secrets to console...)
```js
// Print out configuration settings in debug startup
log.debug('App configuration', config);
```

**Review by jujaga — CHANGES_REQUESTED** [review]

**Review by jujaga — APPROVED** [review]

**Review by TimCsaky — APPROVED** [review]
looks good to me!

**Pull request merged**